src/ja/server/scheduler/algorithm.py

```diff
@@ -0,0 +1,115 @@
+from abc import ABC, abstractmethod
+from typing import List, Optional, Tuple
+from ja.common.job import Job
+from ja.server.database.database import ServerDatabase
+from ja.server.database.types.work_machine import WorkMachine
+
+
+class CostFunction(ABC):
+    """
+    A CostFunction is a collection of functions which determine the order in
```

Comment:
Why a collection of functions?
Shouldn't it be just one function per cost function object and multiple cost function objects?

Reply:
I initially had another design but later changed it, and forgot to update comments. It is as you say, a single cost function.